packages/opentelemetry-rca-metrics/src/metric.ts

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import * as metrics from '@opentelemetry/metrics';
+import * as enums from './enum';
+
+import { getCpuUsageData, getMemoryData } from './stats/common';
+import { getNetworkData } from './stats/si';
+
+import * as types from './types';
+
+/**
+ * Metrics Collector Configuration
+ */
+export interface MetricsCollectorConfig {
+  logger?: api.Logger;
+  exporter: metrics.MetricExporter;
+  // maximum timeout to wait for stats collection default is 500ms
+  maxTimeoutUpdateMS?: number;
+  // Character to be used to join metrics - default is "."
+  metricNameSeparator?: string;
+  // Name of component
+  name: string;
+  // metric export endpoint
+  url: string;
+  // How often the metrics should be exported
+  interval?: number;
+}
+
+const DEFAULT_INTERVAL = 60 * 1000;
+const DEFAULT_MAX_TIMEOUT_UPDATE_MS = 500;
+const DEFAULT_NAME = 'opentelemetry-metrics-collector';
+const DEFAULT_METRIC_NAME_SEPARATOR = '.';
+
+// default label name to be used to store metric name
+const DEFAULT_KEY = 'name';
+
+/**
+ * Metrics Collector - collects metrics for CPU, Memory, Heap, Network, Event
+ * Loop, Garbage Collector, Heap Space
+ * the default label name for metric name is "name"
+ */
+export class RCAMetrics {
+  private _intervalExport: number | undefined;
+  private _exporter: metrics.MetricExporter;
+  protected _logger: api.Logger | undefined;
+  protected _maxTimeoutUpdateMS: number;
+  protected _meter: metrics.Meter;
+  private _name: string;
+  private _boundCounters: { [key: string]: api.BoundCounter } = {};
+  private _metricNameSeparator: string;
+
+  private _memValueObserver: types.ValueObserverWithObservations | undefined;
+
+  constructor(config: MetricsCollectorConfig) {
+    this._intervalExport =
+      typeof config.interval === 'number' ? config.interval : DEFAULT_INTERVAL;
+    this._exporter = config.exporter;
+    this._logger = config.logger;
+    this._name = config.name || DEFAULT_NAME;
+    this._maxTimeoutUpdateMS =
+      config.maxTimeoutUpdateMS || DEFAULT_MAX_TIMEOUT_UPDATE_MS;
+    this._metricNameSeparator =
+      config.metricNameSeparator || DEFAULT_METRIC_NAME_SEPARATOR;
+    this._meter = new metrics.MeterProvider({
```

Comment:
Why do you not use the global (or configured) meter provider the way tracing instrumentation works? Shouldn't the global meter provider already have exporting configured?